File: redome-front/src/app/shared/vo/pedido.contato.finalizado.vo.ts
import { AcaoPedidoContato } from 'app/shared/enums/acao.pedido.contato';
import { Formulario } from '../classes/formulario';
import { BaseEntidade } from '../base.entidade';
import { IPedidoContatoFinalizadoVo } from './interface.pedido.contato.finalizado.vo';

export class PedidoContatoFinalizadoVo implements IPedidoContatoFinalizadoVo {

    private _contactado: boolean; 
	private _contactadoPorTerceiro: boolean; 
	private _acao: AcaoPedidoContato;
	private _idMotivoStatusDoador: number;
	private _tempoInativacaoTemporaria: number;
	private _formulario: Formulario;
    private _hemocentro: number;
    

    /**
     * Getter contactado
     * @return {boolean}
     */
	public get contactado(): boolean {
		return this._contactado;
	}

    /**
     * Setter contactado
     * @param {boolean} value
     */
	public set contactado(value: boolean) {
		this._contactado = value;
	}

    /**
     * Getter contactadoPorTerceiro
     * @return {boolean}
     */
	public get contactadoPorTerceiro(): boolean {
		return this._contactadoPorTerceiro;
	}

    /**
     * Setter contactadoPorTerceiro
     * @param {boolean} value
     */
	public set contactadoPorTerceiro(value: boolean) {
		this._contactadoPorTerceiro = value;
    }

    /**
     * Getter idMotivoStatusDoador
     * @return {number}
     */
	public get idMotivoStatusDoador(): number {
		return this._idMotivoStatusDoador;
	}

    /**
     * Setter idMotivoStatusDoador
     * @param {number} value
     */
	public set idMotivoStatusDoador(value: number) {
		this._idMotivoStatusDoador = value;
	}

    /**
     * Getter tempoInativacaoTemporaria
     * @return {number}
     */
	public get tempoInativacaoTemporaria(): number {
		return this._tempoInativacaoTemporaria;
	}

    /**
     * Setter tempoInativacaoTemporaria
     * @param {number} value
     */
	public set tempoInativacaoTemporaria(value: number) {
		this._tempoInativacaoTemporaria = value;
	}

    /**
     * Getter formulario
     * @return {Formulario}
     */
	public get formulario(): Formulario {
		return this._formulario;
	}

    /**
     * Setter formulario
     * @param {Formulario} value
     */
	public set formulario(value: Formulario) {
		this._formulario = value;
	}

    /**
     * Getter hemocentro
     * @return {number}
     */
	public get hemocentro(): number {
		return this._hemocentro;
	}

    /**
     * Setter hemocentro
     * @param {number} value
     */
	public set hemocentro(value: number) {
		this._hemocentro = value;
    }

    /**
     * Getter acao
     * @return {AcaoPedidoContato}
     */
	public get acao(): AcaoPedidoContato {
		return this._acao;
	}

    /**
     * Setter acao
     * @param {AcaoPedidoContato} value
     */
	public set acao(value: AcaoPedidoContato) {
		this._acao = value;
    }
    
    toJSON(): any {
        return {
            contactado: this.contactado,
            contactadoPorTerceiro: this.contactadoPorTerceiro,
            acao: this.acao ? this.acao.id: undefined,
            idMotivoStatusDoador: this.idMotivoStatusDoador,
	        tempoInativacaoTemporaria: this.tempoInativacaoTemporaria,
	        formulario: this.formulario,
            hemocentro: this.hemocentro

        };
    }   

}

File: redome/src/main/java/br/org/cancer/modred/vo/PedidoContatoFinalizadoVo.java
package br.org.cancer.modred.vo;

import java.io.Serializable;

import br.org.cancer.modred.model.Formulario;
import br.org.cancer.modred.model.domain.AcaoPedidoContato;

/**
 * Classe value object para facilitar a finalização do pedido de contato.
 * 
 * @author brunosousa
 *
 */
public class PedidoContatoFinalizadoVo implements Serializable {
	
	private static final long serialVersionUID = -3819852902232953497L;
	
	private Boolean contactado; 
	private Boolean contactadoPorTerceiro; 
	private Long acao;
	private Long idMotivoStatusDoador;
	private Long tempoInativacaoTemporaria;
	private Formulario formulario;
	private Long hemocentro;
	
	public PedidoContatoFinalizadoVo() {}
	
	/**
	 * Método construtor com atributos.
	 * 
	 * @param contactado informa se o doador foi contactado.
	 * @param acao - Informa a ação que deverá ser feita.
	 */
	public PedidoContatoFinalizadoVo(Boolean contactado, AcaoPedidoContato acao) {
		super();
		this.contactado = contactado;
		if(acao != null) {
			this.acao = acao.getId();
		}
	}
	/**
	 * @return the contactado
	 */
	public Boolean getContactado() {
		return contactado;
	}
	/**
	 * @param contactado the contactado to set
	 */
	public void setContactado(Boolean contactado) {
		this.contactado = contactado;
	}
	/**
	 * @return the contactadoPorTerceiro
	 */
	public Boolean getContactadoPorTerceiro() {
		return contactadoPorTerceiro;
	}
	/**
	 * @param contactadoPorTerceiro the contactadoPorTerceiro to set
	 */
	public void setContactadoPorTerceiro(Boolean contactadoPorTerceiro) {
		this.contactadoPorTerceiro = contactadoPorTerceiro;
	}
	//	/**
	//	 * @return the acao
	//	 */
	//	public AcaoPedidoContato getAcao() {
	//		return AcaoPedidoContato.valueById(acao);
	//	}
	//	/**
	//	 * @param acao the acao to set
	//	 */
	//	public void setAcao(AcaoPedidoContato acao) {
	//		this.acao = acao.getId();
	//	}
	/**
	 * @return the idMotivoStatusDoador
	 */
	public Long getIdMotivoStatusDoador() {
		return idMotivoStatusDoador;
	}
	/**
	 * @return the acao
	 */
	public Long getAcao() {
		return acao;
	}

	/**
	 * @param acao the acao to set
	 */
	public void setAcao(Long acao) {
		this.acao = acao;
	}

	/**
	 * @param idMotivoStatusDoador the idMotivoStatusDoador to set
	 */
	public void setIdMotivoStatusDoador(Long idMotivoStatusDoador) {
		this.idMotivoStatusDoador = idMotivoStatusDoador;
	}
	/**
	 * @return the tempoInativacaoTemporaria
	 */
	public Long getTempoInativacaoTemporaria() {
		return tempoInativacaoTemporaria;
	}
	/**
	 * @param tempoInativacaoTemporaria the tempoInativacaoTemporaria to set
	 */
	public void setTempoInativacaoTemporaria(Long tempoInativacaoTemporaria) {
		this.tempoInativacaoTemporaria = tempoInativacaoTemporaria;
	}

	/**
	 * @return the formulario
	 */
	public Formulario getFormulario() {
		return formulario;
	}

	/**
	 * @param formulario the formulario to set
	 */
	public void setFormulario(Formulario formulario) {
		this.formulario = formulario;
	}

	/**
	 * @return the hemocentro
	 */
	public Long getHemocentro() {
		return hemocentro;
	}

	/**
	 * @param hemocentro the hemocentro to set
	 */
	public void setHemocentro(Long hemocentro) {
		this.hemocentro = hemocentro;
	}
	
}
